lib/spack/spack/solver/concretize.lp

```diff
@@ -1009,7 +1009,7 @@ error(100, "Compiler {1}@{2} requested for {0} cannot be found. Set install_miss
 
 % propagate flags when compiler match
 can_inherit_flags(Package, Dependency, FlagType)
- :- depends_on(Package, Dependency),
+ :- path(Package, Dependency),
```

Comment:
See https://github.com/spack/spack/pull/38512#discussion_r1240690818

Reply:
Moving this here: 

> Leaving a note from an internal discussion on Slack. It's not advisable to use path here, since computing that fact is very expensive (clingo has to compute every possible path in every possible DAG).